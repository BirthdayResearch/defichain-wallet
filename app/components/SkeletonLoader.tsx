import * as React from 'react'
import { DexSkeletonLoader } from './skeletonLoaders/DexSkeletonLoader'
import { TransactionSkeletonLoader } from './skeletonLoaders/TransactionSkeletonLoader'

interface SkeletonLoaderProp {
  row: number
  screen: SkeletonLoaderScreen
}

export enum SkeletonLoaderScreen {
  'Dex' = 'Dex',
  'Transaction' = 'Transaction'
}

export function SkeletonLoader (prop: SkeletonLoaderProp): JSX.Element {
  const skeletonRow = Array.from(Array(prop.row), (_v, i) => i + 1)

  switch (prop.screen) {
    case SkeletonLoaderScreen.Dex:
      return (
        <>
          {skeletonRow.map(i => (
            <DexSkeletonLoader key={i} />
          ))}
        </>
      )

    case SkeletonLoaderScreen.Transaction:
      return (
        <>
          {skeletonRow.map(i => (
            <TransactionSkeletonLoader key={i} />
          ))}
        </>
      )
  }
}
