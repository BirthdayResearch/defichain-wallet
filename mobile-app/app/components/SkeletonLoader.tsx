import { DexSkeletonLoader } from './skeletonLoaders/DexSkeletonLoader'
import { MnemonicWordSkeletonLoader } from './skeletonLoaders/MnemonicWordSkeletonLoader'
import { TransactionSkeletonLoader } from './skeletonLoaders/TransactionSkeletonLoader'
import { LoanSkeletonLoader } from './skeletonLoaders/LoanSkeletonLoader'
import { AddressSkeletonLoader } from './skeletonLoaders/AddressSkeletonLoader'
import { BrowseAuctionsLoader } from './skeletonLoaders/BrowseAuctionsLoader'
import { VaultSkeletonLoader } from './skeletonLoaders/VaultSkeletonLoader'

interface SkeletonLoaderProp {
  row: number
  screen: SkeletonLoaderScreen
}

export enum SkeletonLoaderScreen {
  'Dex' = 'Dex',
  'Transaction' = 'Transaction',
  'MnemonicWord' = 'MnemonicWord',
  'Loan' = 'Loan',
  'Address' = 'Address',
  'BrowseAuction' = 'BrowseAuction',
  'Vault' = 'Vault'
}

export function SkeletonLoader (prop: SkeletonLoaderProp): JSX.Element {
  const skeletonRow = Array.from(Array(prop.row), (_v, i) => i + 1)
  switch (prop.screen) {
    case SkeletonLoaderScreen.Dex:
      return (
        <>
          {skeletonRow.map(i => (
            <DexSkeletonLoader key={i} />
          ))}
        </>
      )

    case SkeletonLoaderScreen.Transaction:
      return (
        <>
          {skeletonRow.map(i => (
            <TransactionSkeletonLoader key={i} />
          ))}
        </>
      )
    case SkeletonLoaderScreen.MnemonicWord:
      return (
        <>
          {skeletonRow.map(i => (
            <MnemonicWordSkeletonLoader key={i} />
          ))}
        </>
      )
    case SkeletonLoaderScreen.Loan:
      return (
        <>
          {skeletonRow.map(i => (
            <LoanSkeletonLoader key={i} />
          ))}
        </>
      )
    case SkeletonLoaderScreen.Address:
        return (
          <>
            {skeletonRow.map(i => (
              <AddressSkeletonLoader key={i} />
            ))}
          </>
        )
    case SkeletonLoaderScreen.BrowseAuction:
      return (
        <>
          {skeletonRow.map(i => (
            <BrowseAuctionsLoader key={i} />
          ))}
        </>
      )
    case SkeletonLoaderScreen.Vault:
      return (
        <>
          {skeletonRow.map(i => (
            <VaultSkeletonLoader key={i} />
          ))}
        </>
      )
  }
}
