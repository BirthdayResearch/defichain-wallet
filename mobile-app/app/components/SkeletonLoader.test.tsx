import { render } from '@testing-library/react-native'
import { SkeletonLoader, SkeletonLoaderScreen } from './SkeletonLoader'

jest.mock('@shared-contexts/ThemeProvider')

describe('Skeleton loader', () => {
  const skeletonLoaders = [
    SkeletonLoaderScreen.Dex,
    SkeletonLoaderScreen.Address,
    SkeletonLoaderScreen.BrowseAuction,
    SkeletonLoaderScreen.Loan,
    SkeletonLoaderScreen.MnemonicWord,
    SkeletonLoaderScreen.Transaction,
    SkeletonLoaderScreen.Vault
  ]

  skeletonLoaders.forEach((eachLoader) => {
    it(`should match snapshot for ${eachLoader}`, () => {
      const rendered = render(
        <SkeletonLoader
          row={1}
          screen={eachLoader}
        />
      )
      expect(rendered.toJSON()).toMatchSnapshot()
    })
  })
})
