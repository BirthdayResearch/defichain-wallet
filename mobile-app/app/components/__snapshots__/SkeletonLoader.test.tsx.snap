// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton loader should match snapshot for Address 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "paddingBottom": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
      },
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(245, 245, 245, 1)",
        "borderLeftColor": "rgba(245, 245, 245, 1)",
        "borderRightColor": "rgba(245, 245, 245, 1)",
        "borderTopColor": "rgba(245, 245, 245, 1)",
      },
    ]
  }
  testID="address_skeleton_loader"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={35}
    bbWidth="100%"
    focusable={false}
    height={35}
    interval={0.25}
    meetOrSlice={1}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {},
        Object {
          "flex": 0,
          "height": 35,
          "width": "100%",
        },
      ]
    }
    vbHeight={35}
    vbWidth={400}
    width="100%"
  >
    <RNSVGGroup>
      <RNSVGRect
        clipPath="emjcqma-animated-diff"
        fill={
          Array [
            1,
            "emjcqma-diff",
          ]
        }
        height="100%"
        propList={
          Array [
            "fill",
          ]
        }
        width="100%"
        x="0"
        y="0"
      />
      <RNSVGDefs>
        <RNSVGClipPath
          name="emjcqma-animated-diff"
        >
          <RNSVGCircle
            cx="35"
            cy="16"
            r="12"
          />
          <RNSVGRect
            height="20"
            rx="3"
            ry="3"
            width="280"
            x="60"
            y="5"
          />
        </RNSVGClipPath>
        <RNSVGLinearGradient
          gradient={
            Array [
              0,
              -1250325,
              0.5,
              -1,
              1,
              -1250325,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="emjcqma-diff"
          x1="-100%"
          x2="0%"
          y1={0}
          y2={0}
        />
      </RNSVGDefs>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Skeleton loader should match snapshot for BrowseAuction 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "justifyContent": "center",
        "marginBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
      },
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 229, 229, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(229, 229, 229, 1)",
        "borderLeftWidth": 1,
        "borderRightColor": "rgba(229, 229, 229, 1)",
        "borderRightWidth": 1,
        "borderTopColor": "rgba(229, 229, 229, 1)",
        "borderTopWidth": 1,
      },
    ]
  }
  testID="browse_auctions_skeleton_loader"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={186}
    bbWidth="100%"
    focusable={false}
    height={186}
    interval={0.25}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {},
        Object {
          "flex": 0,
          "height": 186,
          "width": "100%",
        },
      ]
    }
    vbHeight={186}
    vbWidth={400}
    width="100%"
  >
    <RNSVGGroup>
      <RNSVGRect
        clipPath="hj7xwom-animated-diff"
        fill={
          Array [
            1,
            "hj7xwom-diff",
          ]
        }
        height="100%"
        propList={
          Array [
            "fill",
          ]
        }
        width="100%"
        x="0"
        y="0"
      />
      <RNSVGDefs>
        <RNSVGClipPath
          name="hj7xwom-animated-diff"
        >
          <RNSVGCircle
            cx="21"
            cy="27"
            r="16"
          />
          <RNSVGRect
            height="26"
            rx="3"
            ry="3"
            width="120"
            x="46"
            y="12"
          />
          <RNSVGCircle
            cx="347"
            cy="21"
            r="8"
          />
          <RNSVGCircle
            cx="365"
            cy="21"
            r="8"
          />
          <RNSVGCircle
            cx="382"
            cy="21"
            r="8"
          />
          <RNSVGRect
            height="12"
            rx="3"
            ry="3"
            width="120"
            x="8"
            y="56"
          />
          <RNSVGRect
            height="12"
            rx="3"
            ry="3"
            width="110"
            x="282"
            y="56"
          />
          <RNSVGRect
            height="12"
            rx="3"
            ry="3"
            width="132"
            x="8"
            y="81"
          />
          <RNSVGRect
            height="12"
            rx="3"
            ry="3"
            width="118"
            x="272"
            y="81"
          />
          <RNSVGRect
            height="12"
            rx="3"
            ry="3"
            width="170"
            x="8"
            y="105"
          />
          <RNSVGRect
            height="12"
            rx="3"
            ry="3"
            width="140"
            x="252"
            y="105"
          />
          <RNSVGRect
            height="12"
            rx="3"
            ry="3"
            width="384"
            x="8"
            y="132"
          />
          <RNSVGRect
            height="20"
            rx="3"
            ry="3"
            width="100"
            x="8"
            y="162"
          />
          <RNSVGRect
            height="20"
            rx="3"
            ry="3"
            width="130"
            x="118"
            y="162"
          />
        </RNSVGClipPath>
        <RNSVGLinearGradient
          gradient={
            Array [
              0,
              -1250325,
              0.5,
              -1,
              1,
              -1250325,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="hj7xwom-diff"
          x1="-100%"
          x2="0%"
          y1={0}
          y2={0}
        />
      </RNSVGDefs>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Skeleton loader should match snapshot for Dex 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "borderTopWidth": 1,
        "justifyContent": "center",
        "marginBottom": 8,
        "paddingBottom": 16,
        "paddingTop": 16,
      },
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 229, 229, 1)",
        "borderLeftColor": "rgba(229, 229, 229, 1)",
        "borderRightColor": "rgba(229, 229, 229, 1)",
        "borderTopColor": "rgba(229, 229, 229, 1)",
      },
    ]
  }
  testID="dex_skeleton_loader"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={190}
    bbWidth="100%"
    focusable={false}
    height={190}
    interval={0.25}
    meetOrSlice={1}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {},
        Object {
          "flex": 0,
          "height": 190,
          "width": "100%",
        },
      ]
    }
    vbHeight={190}
    vbWidth={328}
    width="100%"
  >
    <RNSVGGroup>
      <RNSVGRect
        clipPath="notj4tn-animated-diff"
        fill={
          Array [
            1,
            "notj4tn-diff",
          ]
        }
        height="100%"
        propList={
          Array [
            "fill",
          ]
        }
        width="100%"
        x="0"
        y="0"
      />
      <RNSVGDefs>
        <RNSVGClipPath
          name="notj4tn-animated-diff"
        >
          <RNSVGCircle
            cx="30"
            cy="20"
            r="12"
          />
          <RNSVGRect
            height="16"
            rx="5"
            ry="5"
            width="110"
            x="60"
            y="13"
          />
          <RNSVGRect
            height="12"
            rx="4"
            ry="4"
            width="100"
            x="16"
            y="46"
          />
          <RNSVGRect
            height="14"
            rx="4"
            ry="4"
            width="60"
            x="16"
            y="64"
          />
          <RNSVGRect
            height="12"
            rx="4"
            ry="4"
            width="100"
            x="166"
            y="46"
          />
          <RNSVGRect
            height="14"
            rx="4"
            ry="4"
            width="60"
            x="166"
            y="64"
          />
          <RNSVGRect
            height="12"
            rx="4"
            ry="4"
            width="100"
            x="16"
            y="96"
          />
          <RNSVGRect
            height="14"
            rx="4"
            ry="4"
            width="60"
            x="16"
            y="114"
          />
          <RNSVGRect
            height="12"
            rx="4"
            ry="4"
            width="100"
            x="166"
            y="96"
          />
          <RNSVGRect
            height="14"
            rx="4"
            ry="4"
            width="60"
            x="166"
            y="114"
          />
          <RNSVGRect
            height="30"
            rx="4"
            ry="4"
            width="144"
            x="16"
            y="145"
          />
          <RNSVGRect
            height="30"
            rx="4"
            ry="4"
            width="80"
            x="168"
            y="145"
          />
          <RNSVGRect
            height="30"
            rx="4"
            ry="4"
            width="32"
            x="280"
            y="145"
          />
        </RNSVGClipPath>
        <RNSVGLinearGradient
          gradient={
            Array [
              0,
              -1250325,
              0.5,
              -1,
              1,
              -1250325,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="notj4tn-diff"
          x1="-100%"
          x2="0%"
          y1={0}
          y2={0}
        />
      </RNSVGDefs>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Skeleton loader should match snapshot for Loan 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginLeft": 8,
        "marginRight": 8,
        "marginTop": 8,
        "paddingBottom": 4,
        "paddingTop": 4,
      },
      Object {
        "backgroundColor": "rgba(245, 245, 245, 1)",
      },
    ]
  }
  testID="loan_skeleton_loader"
>
  <View
    style={
      Array [
        Array [
          Object {
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          Object {
            "flexBasis": "47%",
          },
        ],
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderBottomColor": "rgba(229, 229, 229, 1)",
          "borderBottomWidth": 1,
          "borderLeftColor": "rgba(229, 229, 229, 1)",
          "borderLeftWidth": 1,
          "borderRightColor": "rgba(229, 229, 229, 1)",
          "borderRightWidth": 1,
          "borderTopColor": "rgba(229, 229, 229, 1)",
          "borderTopWidth": 1,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={144}
      bbWidth="100%"
      focusable={false}
      height={144}
      interval={0.25}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {},
          Object {
            "flex": 0,
            "height": 144,
            "width": "100%",
          },
        ]
      }
      vbHeight={144}
      vbWidth={163}
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGRect
          clipPath="889hpl-animated-diff"
          fill={
            Array [
              1,
              "889hpl-diff",
            ]
          }
          height="100%"
          propList={
            Array [
              "fill",
            ]
          }
          width="100%"
          x="0"
          y="0"
        />
        <RNSVGDefs>
          <RNSVGClipPath
            name="889hpl-animated-diff"
          >
            <RNSVGRect
              height="14"
              rx="6"
              ry="6"
              width="90"
              x="48"
              y="24"
            />
            <RNSVGRect
              height="12"
              rx="5"
              ry="5"
              width="51"
              x="13"
              y="56"
            />
            <RNSVGRect
              height="12"
              rx="6"
              ry="6"
              width="96"
              x="13"
              y="75"
            />
            <RNSVGCircle
              cx="26"
              cy="31"
              r="13"
            />
            <RNSVGRect
              height="12"
              rx="6"
              ry="6"
              width="51"
              x="13"
              y="100"
            />
            <RNSVGRect
              height="12"
              rx="6"
              ry="6"
              width="96"
              x="13"
              y="118"
            />
          </RNSVGClipPath>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1250325,
                0.5,
                -1,
                1,
                -1250325,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="889hpl-diff"
            x1="-100%"
            x2="0%"
            y1={0}
            y2={0}
          />
        </RNSVGDefs>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      Array [
        Array [
          Object {
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          Object {
            "flexBasis": "47%",
          },
        ],
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderBottomColor": "rgba(229, 229, 229, 1)",
          "borderBottomWidth": 1,
          "borderLeftColor": "rgba(229, 229, 229, 1)",
          "borderLeftWidth": 1,
          "borderRightColor": "rgba(229, 229, 229, 1)",
          "borderRightWidth": 1,
          "borderTopColor": "rgba(229, 229, 229, 1)",
          "borderTopWidth": 1,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={144}
      bbWidth="100%"
      focusable={false}
      height={144}
      interval={0.25}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {},
          Object {
            "flex": 0,
            "height": 144,
            "width": "100%",
          },
        ]
      }
      vbHeight={144}
      vbWidth={163}
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGRect
          clipPath="zrsylmi-animated-diff"
          fill={
            Array [
              1,
              "zrsylmi-diff",
            ]
          }
          height="100%"
          propList={
            Array [
              "fill",
            ]
          }
          width="100%"
          x="0"
          y="0"
        />
        <RNSVGDefs>
          <RNSVGClipPath
            name="zrsylmi-animated-diff"
          >
            <RNSVGRect
              height="14"
              rx="6"
              ry="6"
              width="90"
              x="48"
              y="24"
            />
            <RNSVGRect
              height="12"
              rx="5"
              ry="5"
              width="51"
              x="13"
              y="56"
            />
            <RNSVGRect
              height="12"
              rx="6"
              ry="6"
              width="96"
              x="13"
              y="75"
            />
            <RNSVGCircle
              cx="26"
              cy="31"
              r="13"
            />
            <RNSVGRect
              height="12"
              rx="6"
              ry="6"
              width="51"
              x="13"
              y="100"
            />
            <RNSVGRect
              height="12"
              rx="6"
              ry="6"
              width="96"
              x="13"
              y="118"
            />
          </RNSVGClipPath>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1250325,
                0.5,
                -1,
                1,
                -1250325,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="zrsylmi-diff"
            x1="-100%"
            x2="0%"
            y1={0}
            y2={0}
          />
        </RNSVGDefs>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Skeleton loader should match snapshot for MnemonicWord 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "width": "100%",
      },
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 229, 229, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(229, 229, 229, 1)",
        "borderRightColor": "rgba(229, 229, 229, 1)",
        "borderTopColor": "rgba(229, 229, 229, 1)",
      },
    ]
  }
  testID="mnemonic_word_skeleton_loader"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={47}
    bbWidth="100%"
    focusable={false}
    height={47}
    interval={0.25}
    meetOrSlice={1}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {},
        Object {
          "flex": 0,
          "height": 47,
          "width": "100%",
        },
      ]
    }
    vbHeight={47}
    vbWidth={400}
    width="100%"
  >
    <RNSVGGroup>
      <RNSVGRect
        clipPath="i7pe4yh-animated-diff"
        fill={
          Array [
            1,
            "i7pe4yh-diff",
          ]
        }
        height="100%"
        propList={
          Array [
            "fill",
          ]
        }
        width="100%"
        x="0"
        y="0"
      />
      <RNSVGDefs>
        <RNSVGClipPath
          name="i7pe4yh-animated-diff"
        >
          <RNSVGRect
            height="33"
            rx="3"
            ry="3"
            width="290"
            x="70"
            y="3"
          />
          <RNSVGCircle
            cx="20"
            cy="20"
            matrix={
              Array [
                1,
                0,
                -0,
                1,
                20,
                0,
              ]
            }
            r="20"
          />
        </RNSVGClipPath>
        <RNSVGLinearGradient
          gradient={
            Array [
              0,
              -1250325,
              0.5,
              -1,
              1,
              -1250325,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="i7pe4yh-diff"
          x1="-100%"
          x2="0%"
          y1={0}
          y2={0}
        />
      </RNSVGDefs>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Skeleton loader should match snapshot for Transaction 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "width": "100%",
      },
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 229, 229, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(229, 229, 229, 1)",
        "borderRightColor": "rgba(229, 229, 229, 1)",
        "borderTopColor": "rgba(229, 229, 229, 1)",
      },
    ]
  }
  testID="transaction_skeleton_loader"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={47}
    bbWidth="100%"
    focusable={false}
    height={47}
    interval={0.25}
    meetOrSlice={1}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {},
        Object {
          "flex": 0,
          "height": 47,
          "width": "100%",
        },
      ]
    }
    vbHeight={47}
    vbWidth={344}
    width="100%"
  >
    <RNSVGGroup>
      <RNSVGRect
        clipPath="22v9wz-animated-diff"
        fill={
          Array [
            1,
            "22v9wz-diff",
          ]
        }
        height="100%"
        propList={
          Array [
            "fill",
          ]
        }
        width="100%"
        x="0"
        y="0"
      />
      <RNSVGDefs>
        <RNSVGClipPath
          name="22v9wz-animated-diff"
        >
          <RNSVGCircle
            cx="17"
            cy="23"
            r="10"
          />
          <RNSVGRect
            height="15"
            width="100"
            x="44"
            y="7"
          />
          <RNSVGRect
            height="10"
            width="100"
            x="44"
            y="29"
          />
          <RNSVGRect
            height="15"
            width="85"
            x="250"
            y="16"
          />
        </RNSVGClipPath>
        <RNSVGLinearGradient
          gradient={
            Array [
              0,
              -1250325,
              0.5,
              -1,
              1,
              -1250325,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="22v9wz-diff"
          x1="-100%"
          x2="0%"
          y1={0}
          y2={0}
        />
      </RNSVGDefs>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Skeleton loader should match snapshot for Vault 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 8,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 229, 229, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(229, 229, 229, 1)",
        "borderLeftWidth": 1,
        "borderRightColor": "rgba(229, 229, 229, 1)",
        "borderRightWidth": 1,
        "borderTopColor": "rgba(229, 229, 229, 1)",
        "borderTopWidth": 1,
      },
    ]
  }
  testID="vault_skeleton_loader"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={130}
    bbWidth="100%"
    focusable={false}
    height={130}
    interval={0.25}
    meetOrSlice={1}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {},
        Object {
          "flex": 0,
          "height": 130,
          "width": "100%",
        },
      ]
    }
    vbHeight={130}
    vbWidth={328}
    width="100%"
  >
    <RNSVGGroup>
      <RNSVGRect
        clipPath="8dfb1rl-animated-diff"
        fill={
          Array [
            1,
            "8dfb1rl-diff",
          ]
        }
        height="100%"
        propList={
          Array [
            "fill",
          ]
        }
        width="100%"
        x="0"
        y="0"
      />
      <RNSVGDefs>
        <RNSVGClipPath
          name="8dfb1rl-animated-diff"
        >
          <RNSVGRect
            height="16"
            width="60"
            x="0"
            y="13"
          />
          <RNSVGRect
            height="16"
            width="60"
            x="70"
            y="13"
          />
          <RNSVGRect
            height="16"
            width="240"
            x="0"
            y="33"
          />
          <RNSVGRect
            height="15"
            width="150"
            x="0"
            y="53"
          />
          <RNSVGRect
            height="15"
            width="130"
            x="0"
            y="90"
          />
          <RNSVGRect
            height="15"
            width="140"
            x="220"
            y="90"
          />
          <RNSVGRect
            height="15"
            width="130"
            x="0"
            y="115"
          />
          <RNSVGRect
            height="15"
            width="140"
            x="220"
            y="115"
          />
        </RNSVGClipPath>
        <RNSVGLinearGradient
          gradient={
            Array [
              0,
              -1250325,
              0.5,
              -1,
              1,
              -1250325,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="8dfb1rl-diff"
          x1="-100%"
          x2="0%"
          y1={0}
          y2={0}
        />
      </RNSVGDefs>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
